renderdoc/driver/metal/metal_serialise.cpp

```diff
@@ -221,8 +249,83 @@ void DoSerialise(SerialiserType &ser, RDMTL::RenderPipelineDescriptor &el)
   SERIALISE_MEMBER(maxFragmentCallStackDepth);
 }
 
+template <typename SerialiserType>
+void DoSerialise(SerialiserType &ser, RDMTL::RenderPassAttachmentDescriptor &el)
+{
+  SERIALISE_MEMBER(texture);
+  SERIALISE_MEMBER(level);
+  SERIALISE_MEMBER(slice);
+  SERIALISE_MEMBER(depthPlane);
+  SERIALISE_MEMBER(resolveTexture);
+  SERIALISE_MEMBER(resolveLevel);
+  SERIALISE_MEMBER(resolveSlice);
+  SERIALISE_MEMBER(resolveDepthPlane);
+  SERIALISE_MEMBER(loadAction);
+  SERIALISE_MEMBER(storeAction);
+  SERIALISE_MEMBER(storeActionOptions);
+};
+
+template <typename SerialiserType>
+void DoSerialise(SerialiserType &ser, RDMTL::RenderPassColorAttachmentDescriptor &el)
+{
+  DoSerialise(ser, (RDMTL::RenderPassAttachmentDescriptor &)el);
+  SERIALISE_MEMBER(clearColor);
+};
+
+template <typename SerialiserType>
+void DoSerialise(SerialiserType &ser, RDMTL::RenderPassDepthAttachmentDescriptor &el)
+{
+  DoSerialise(ser, (RDMTL::RenderPassAttachmentDescriptor &)el);
+  SERIALISE_MEMBER(clearDepth);
+  SERIALISE_MEMBER(depthResolveFilter);
+};
+
+template <typename SerialiserType>
+void DoSerialise(SerialiserType &ser, RDMTL::RenderPassStencilAttachmentDescriptor &el)
+{
+  DoSerialise(ser, (RDMTL::RenderPassAttachmentDescriptor &)el);
+  SERIALISE_MEMBER(clearStencil);
+  SERIALISE_MEMBER(stencilResolveFilter);
+};
+
+template <typename SerialiserType>
+void DoSerialise(SerialiserType &ser, RDMTL::RenderPassSampleBufferAttachmentDescriptor &el)
+{
+  // TODO: will MTL::CounterSampleBuffer need to be a wrapped resource
```

Comment:
Minor note to answer this question - if you have to ask the answer is usually yes. I've occasionally opted not to wrap something and I tend to end up needing to wrap it eventually.